src/Farmer/Builders/Builders.Storage.fs

```diff
@@ -344,6 +348,25 @@ type StorageAccountBuilder() =
                         IpRules = allowIp :: existingAcl.IpRules
                     } |> Some
             }
+    /// Restrict access to this storage account to the private endpoints and azure services.
+    [<CustomOperation "restrict_to_azure_services">]
+    member _.RestrictToAzureServices(state:StorageAccountConfig, bypass:NetworkRuleSetBypass list) =
+        match state.NetworkAcls with
+        | None ->
+            { state with
+                NetworkAcls =
+                    { Bypass = set bypass
+                      VirtualNetworkRules = []
+                      IpRules = []
+                      DefaultAction = RuleAction.Deny } |> Some
+            }
+        | Some existingAcl ->
```

Comment:
Should we replace any existing ACLs with this ? So only the 'bypass' list is allowed but not any IPs etc ?
`{ 
    Bypass = set bypass
    VirtualNetworkRules = []
    IpRules = []
    DefaultAction = RuleAction.Deny 
}`

Reply:
I don't think so, restricting access to particular subnets/cidr ranges and allowing azure services / metrics / etc to bypass those rules seems like a genuine use-case.

Having said that, having looked at the code for the other `restrict_to_xxx` methods, they're all additive, so I've made this additive as well. (See the tests)